### PR TITLE
Resprite Hunter loot and make it self destruct on death

### PR DESCRIPTION
**Pull request by flappybatpal**
[BALANCE]

## About the PR 
Remove all text references to a certain movie.
Make hunter spear, cloaking device and gun self destruct on hunter's death.
Resprites hunter's gun and cloak.
Change the type for laser rifles to plasma rifle to better represent their projectile.
Rename laser rifles to plasma rifle.

image
image

## Why's this needed? 
If you manage to loot a hunter you get a butcher knife equivalent spear, recharging laser gun, cloaking device and thermal goggles/voice changer mask. This is 19TC worth of traitor gear for the spear, cloak and gun making them a gold mine for valid hunters.
Make the hunter's appearance more unique.

## Changelog

```changelog
(u)Flappybat and Flaborized and Virvatuli.
(*)Hunter laser rifle, spear and cloaking device self destruct on the hunter's death. These items have been renamed to make them stand out more.
(*)Hunter's laser rifle and cloaking device have new sprites by Flaborized!
(*)Laser rifles are renamed to plasma rifle to reflect their projectile and have a new sound by Virvatuli.
```

### CODE REVIEW

**Comment by flappybatpal** [comment]
I'd like to nerf the spear but don't see a good way to do that due to the inheritance.

File: code/obj/storage/loot_crates.dm [hunk 1]
```diff
@@ -180,7 +180,7 @@
 					picker = rand(1,10)
 					switch(picker)
 						if(1)
-							items += pick(/obj/item/gun/energy/laser_gun/pred)
+							items += pick(/obj/item/gun/energy/laser_gun)
```
Comment (Zonespace27) [inline code/obj/storage/loot_crates.dm:183]: is knocking this down to the regular one from an advanced gun on purpose?

**Comment by virvatuli-gh** [comment]
Equipment should self-destruct when the Hunter self-destructs, preferably violently 👀

File: code/obj/storage/loot_crates.dm (hunk 1 above)
Comment (flappybatpal) [reply code/obj/storage/loot_crates.dm]: Whoops

**Comment by flappybatpal** [comment]
> Equipment should self-destruct when the Hunter self-destructs, preferably violently 👀

That's an idea that could nicely extend to all their gear and is probably quite easy with signals.
Depends what the best solution is? I don't think they could reasonably be exploited to keep hunters alive, although friendly hunters losing their gear could become a thing. Generally though they end up killed with their own gear after getting stunned.

**Comment by Tarmunora** [comment]
I think that having the gear self-destruct would be preferable to just being paperweights for non-hunters

**Comment by virvatuli-gh** [comment]
> Typo in changelog, also if theres two hunters in a round and one hunter dies they wont cause both sets of hunter gear to explode right?

This is part of the stuff that's currently being ironed out on the dev side of things as we figure out was to make the Hunter antag more interesting in general.

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**